From 434dfe6d1095fab671eba1518823603d0f071e2c Mon Sep 17 00:00:00 2001
From: Andy Whitcroft <apw@canonical.com>
Date: Fri, 23 Jan 2009 12:05:52 +0000
Subject: [PATCH] UBUNTU: allow us to build default configs for automated builds

Allow us to request an automated update to the configs taking whatever
the default answers are.  This allows us to automate builds of mainline
kernels using our build infrastructure and configs.

Signed-off-by: Andy Whitcroft <apw@canonical.com>
---
 debian/rules.d/1-maintainer.mk |   10 ++++++++++
 debian/scripts/misc/oldconfig  |   12 ++++++++----
 2 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/debian/rules.d/1-maintainer.mk b/debian/rules.d/1-maintainer.mk
index 0c1bf38..d63a045 100644
--- a/debian/rules.d/1-maintainer.mk
+++ b/debian/rules.d/1-maintainer.mk
@@ -10,6 +10,8 @@ help:
 	@echo
 	@echo "  updateconfigs   : Update debian/config/*"
 	@echo
+	@echo "  defaultconfigs  : Update debian/config/*"
+	@echo
 	@echo "  editconfigs     : Update debian/config/* interactively"
 	@echo
 	@echo "  printchanges    : Print the current changelog entries (from git)"
@@ -39,6 +41,14 @@ updateconfigs:
 	done
 	rm -rf build
 
+defaultconfigs:
+	dh_testdir
+	@for arch in $(ARCH_CONFIGS); do	\
+		yes '' | \
+		$(SHELL) debian/scripts/misc/oldconfig $$arch oldconfig; \
+	done
+	rm -rf build
+
 editconfigs:
 	dh_testdir
 	@for arch in $(ARCH_CONFIGS); do	\
diff --git a/debian/scripts/misc/oldconfig b/debian/scripts/misc/oldconfig
index 9dfdc9a..dfbeff5 100755
--- a/debian/scripts/misc/oldconfig
+++ b/debian/scripts/misc/oldconfig
@@ -8,12 +8,16 @@ fi
 
 
 # One arg, and that's it. Just pass an architecture
-if [ $# -ne 1 ]; then
-	echo "Usage: $0 <arch>" 1>&2
+if [ $# -ne 1 -a $# -ne 2 ]; then
+	echo "Usage: $0 <arch> [<operation>]" 1>&2
 	exit 1
 fi
 
 arch="$1"
+what="silentoldconfig"
+if [ $# -eq 2 ]; then
+	what="$2"
+fi
 
 case "$arch" in
 	amd64)	kernarch="x86_64"	;;
@@ -48,11 +52,11 @@ fi
 test -d build || mkdir build
 cd build
 for config in $configs; do
-	echo "Running silentoldconfig for $config ... "
+	echo "Running $what for $config ... "
 
 	cat $confdir/$config > .config
 
-	make -C ../ O=`pwd` silentoldconfig ARCH=$kernarch
+	make -C ../ O=`pwd` "$what" ARCH=$kernarch
 
 	cat .config > $confdir/$config
 done
-- 
1.6.1.2.419.g0d87e

